code/game/machinery/rechargestation.dm

```diff
@@ -138,6 +153,17 @@
 			return
 		else if(default_part_replacement(user, O))
 			return
+		if(istype(O, /obj/item/weapon/reagent_containers) && O.reagents)
+			for (var/datum/reagent/R in O.reagents.reagent_list)

```

Comment:
similar to the comment above. `for (var/a in beaker.reagents.reagent_list)` and then just have line `var/datum/reagent/R = a` right below for loop.  this will save in-built istype check that for each loop has in byond.